### PR TITLE
Feat: Auto Map V1

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
Adding mira call to auto map feature

To Test:

1. Grab a dataset, make sure that it is enriched or enrich it.
2. Grab a model make sure this is enriched as well
3. Open either calibration node and click automap

Sample:
.../projects/5/workflows/811fca52-31ad-4fee-bd2c-5db972af5c14

https://github.com/DARPA-ASKEM/terarium/assets/17088680/75f92cb1-b033-48d8-9996-a96420819bdc

Things to note:
This is using pairwise comparison rather than cluster comparison. Ill start a conversation about this should we care to ask MIRA to change this

### CODE REVIEW

File: packages/client/hmi-client/src/services/calibrate-workflow.ts [hunk 1]
```diff
@@ -108,3 +116,29 @@ export const renderLossGraph = (
...
+			// No direct string match, check grounding keys (if they exist)
+			else if (
+				modelOption.grounding?.identifiers &&
+				datasetOption?.metadata?.groundings?.identifiers
+			) {
+				const modelTemp = Object.entries(modelOption.grounding.identifiers);
+				const datasetTemp = Object.entries(datasetOption?.metadata?.groundings?.identifiers);
+				const modelGroundingList = modelTemp.map((ele) => ele.join(':'));
+				const dataGroundingList = datasetTemp.map((ele) => ele.join(':'));
+				console.log(modelGroundingList);
```
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/services/calibrate-workflow.ts:137]: Need to utilize MIRA results here rather than logging them of course

File: packages/client/hmi-client/src/services/calibrate-workflow.ts [hunk 2]
```diff
@@ -116,27 +117,42 @@ export const renderLossGraph = (
...
+				datasetOptions[j]?.metadata?.groundings?.identifiers
 			) {
-				const datasetKeys = Object.keys(datasetOption.metadata.groundings.identifiers);
-				Object.keys(modelOption.grounding.identifiers).forEach((modelGrounding) => {
-					if (datasetKeys.includes(modelGrounding)) {
-						result.push({ modelVariable: modelOption.id, datasetVariable: datasetOption.name });
+				const modelTemp = Object.entries(modelOptions[i].grounding?.identifiers);
+				const datasetTemp = Object.entries(datasetOptions[j].metadata?.groundings?.identifiers);
+				const modelGroundingList = modelTemp.map((ele) => ele.join(':'));
+				const dataGroundingList = datasetTemp.map((ele) => ele.at(0) as string);
```
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/services/calibrate-workflow.ts:140]: confirming this is always correct. KMW may not provide such consistent formatting

File: packages/client/hmi-client/src/services/concept.ts [hunk 3]
```diff
@@ -51,4 +51,18 @@ async function getCuriesEntities(curies: Array<string>): Promise<Array<DKG> | nu
 	}
 }
 
-export { getCuriesEntities, getFacets };
+async function getEntitySimilarity(
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/concept.ts:54]: Could you add a small JSDoc to explain what this does.

File: packages/client/hmi-client/src/services/calibrate-workflow.ts [hunk 4]
```diff
@@ -116,27 +117,67 @@ export const renderLossGraph = (
...
-				}); // End for each grounding key
+	const allModelGroundings: string[] = [];
+	const allDataGroundings: string[] = [];
+	const acceptableDistance = 0.7;
+	// Get all model groundings
+	modelOptions.forEach((state) => {
+		if (state.grounding?.identifiers) {
+			const modelTemp = Object.entries(state.grounding?.identifiers);
+			const modelGroundingList = modelTemp.map((ele) => ele.join(':'));
+			modelGroundingList.forEach((ele) => allModelGroundings.push(ele));
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/calibrate-workflow.ts:133]: Why not chaining those calls?

```suggestion
		Object.entries(state.grounding?.identifiers)
		    .map((ele) => ele.join(':'))
		    .forEach(allModelGroundings.push);
```

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/services/calibrate-workflow.ts (hunk 4 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/services/calibrate-workflow.ts:133]: Good call
Will do

File: packages/client/hmi-client/src/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss.vue [hunk 5]
```diff
@@ -31,6 +36,18 @@
...
 							</template>
 						</Column>
+						<Column field="deleteRow">
+							<template #header>
+								<span class="column-header"></span>
+							</template>
+							<template #body="{ index }">
+								<Button
+									class="p-button-sm p-button-text"
+									label="Delete Row"
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss.vue:46]: The interface looks a tad cramped in your video, maybe we can just shorten this to "delete" instead of "delete row"?

File: packages/client/hmi-client/src/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss.vue [hunk 6]
```diff
@@ -356,6 +374,23 @@ function addMapping() {
...
+function deleteMapping() {
+	mapping.value = [{ modelVariable: '', datasetVariable: '' }];
+
+	const state = _.cloneDeep(props.node.state);
+	state.mapping = mapping.value;
+
+	emit('update-state', state);
+}
+
+function deleteMapRow(index) {
```
Comment (mwdchang) [inline packages/client/hmi-client/src/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss.vue:386]: Typing? `index: number`?

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss.vue]: Thanks for catching this

**Review by mwdchang — APPROVED** [review]
